staging/src/k8s.io/sample-apiserver/pkg/registry/wardle/flunder/strategy.go

```diff
@@ -63,31 +64,99 @@ type flunderStrategy struct {
 	names.NameGenerator
 }
 
+// NamespaceScoped returns true because all Flunders need to be within a namespace.
 func (flunderStrategy) NamespaceScoped() bool {
 	return true
 }
 
+// PrepareForCreate clears the status of a Flunder before creation.
 func (flunderStrategy) PrepareForCreate(ctx genericapirequest.Context, obj runtime.Object) {
+	flunder := obj.(*wardle.Flunder)
+
+	// create cannot set status
+	flunder.Status = wardle.FlunderStatus{}
+	flunder.Generation = 1
 }
 
+// PrepareForUpdate clears fields that are not allowed to be set by end users on update.
 func (flunderStrategy) PrepareForUpdate(ctx genericapirequest.Context, obj, old runtime.Object) {
+	newFlunder := obj.(*wardle.Flunder)
+	oldFlunder := old.(*wardle.Flunder)
+
+	// Update is not allowed to set status
+	newFlunder.Status = oldFlunder.Status
+
+	// Any changes to the spec increment the generation number, any changes to the
+	// status should reflect the generation number of the corresponding object.
+	// See metav1.ObjectMeta description for more information on Generation.
+	if !apiequality.Semantic.DeepEqual(oldFlunder.Spec, newFlunder.Spec) {
+		newFlunder.Generation = oldFlunder.Generation + 1
+	}
 }
 
+// Validate validates a new Flunder.
 func (flunderStrategy) Validate(ctx genericapirequest.Context, obj runtime.Object) field.ErrorList {
 	return field.ErrorList{}
 }
 
+// AllowCreateOnUpdate is false for Flunder; this means POST is needed to create one.
 func (flunderStrategy) AllowCreateOnUpdate() bool {
 	return false
 }
 
+// AllowUnconditionalUpdate is the default update policy for Flunder objects.
 func (flunderStrategy) AllowUnconditionalUpdate() bool {
 	return false
 }
 
+// Canonicalize normalizes the object after validation.
 func (flunderStrategy) Canonicalize(obj runtime.Object) {
 }
 
+// ValidateUpdate is the default update validation for an end user.
 func (flunderStrategy) ValidateUpdate(ctx genericapirequest.Context, obj, old runtime.Object) field.ErrorList {
 	return field.ErrorList{}
 }
+
+type flunderStatusStrategy struct {
+	runtime.ObjectTyper
+	names.NameGenerator
+}
+
+func NewFlunderStatusStrategy(typer runtime.ObjectTyper) flunderStatusStrategy {
+	return flunderStatusStrategy{typer, names.SimpleNameGenerator}
+}
+
+func (flunderStatusStrategy) NamespaceScoped() bool {
+	return false
+}
+
+func (flunderStatusStrategy) AllowCreateOnUpdate() bool {
+	return false
+}
+
+func (flunderStatusStrategy) AllowUnconditionalUpdate() bool {
+	return false
+}
+
+func (flunderStatusStrategy) Canonicalize(obj runtime.Object) {
+}
+
+// PrepareForUpdate clears fields that are not allowed to be set by end users on update of status.
+func (flunderStatusStrategy) PrepareForUpdate(ctx genericapirequest.Context, obj, old runtime.Object) {
+	newObj := obj.(*wardle.Flunder)
+	oldObj := old.(*wardle.Flunder)
+	newObj.Spec = oldObj.Spec
+
+	// Status updates are for only for updating status, not objectmeta.
+	newObj.Labels = oldObj.Labels
```

Comment:
Let's bring in https://github.com/kubernetes/kubernetes/pull/45552/files for this.  @sttts @kubernetes/sig-api-machinery-misc

Reply:
https://github.com/kubernetes/kubernetes/pull/45552/files is more robust.  We can start it scoped to this resource as a demonstration.

Reply:
Take two for https://github.com/kubernetes/kubernetes/pull/45552: https://github.com/kubernetes/kubernetes/pull/59038